libs/wingc/src/type_check.rs

```diff
@@ -2133,12 +2134,51 @@ impl<'a> TypeChecker<'a> {
 							// Left argument must be an optional type
 							if !ltype.is_option() {
 								self.spanned_error(left, format!("Expected optional type, found \"{}\"", ltype));
-								(ltype, phase)
-							} else {
-								// Right argument must be a subtype of the inner type of the left argument
-								let inner_type = *ltype.maybe_unwrap_option();
-								self.validate_type(rtype, inner_type, right);
-								(inner_type, phase)
```

Comment:
I kind of see the logic in this straight forward "unwrap" implementation. Perhaps the original issue with #1875 is just a grammar right associativity thing.

Reply:
I just added a grammar test - [`fd14377` (#5984)](https://github.com/winglang/wing/pull/5984/commits/fd1437790e57a437b0448a2e4bc8e5bea59194b1)

If I look at the output, I think it's parsing (a ?? b ?? c) as ((a ?? b) ?? c) which I believe is correct

Reply:
On second thought, I think the operation is may be associative so the grouping of parens may not really matter all that much.

Reply:
I updated the PR description with a better explanation of the fixes

Reply:
note Rusts [`unwapr_or`](https://doc.rust-lang.org/std/option/enum.Option.html#method.unwrap_or), the result is always `T` and never an `Option<T>`.

Reply:
Ah I think I see what you mean now... yea changing the associativity should fix the original issue.

Reply:
What's wrong with `a ?? b ?? c` -> `(a ?? b) ?? c`? Just a desire to avoid `??` resulting in an optional? 
To me this seems to solve the issue and also addresses https://github.com/winglang/wing/issues/5944. It also matches how JS works (although it's not necessary to match that)

Reply:
I added more context for the change in #5944. In the net this seems like a good change so let's try it